text/trace/0235-sampling-threshold-in-trace-state.md

```diff
@@ -0,0 +1,87 @@
+* Else the value corresponding to the `th` key should be interpreted as above.
+
+Sampling Decisions SHOULD be propagated by setting the value of the `th` key in the Tracestate header according to the above.
+
+In the case of a downstream sampler that is attempting to reduce the volume of traffic, the sampler MAY modify the `th` header.
+Note that the `T` value may be reduced by a downstream sampler, but should never be increased.
+
+A sampler SHALL introduce an R value to a trace that does not include one and does not have the `Random` trace-id flag set. It SHALL use the `rv` key for this purpose. A sampler SHALL NOT modify an existing R value or trace-id.
+
+## Internal details
+
+The trace state header SHOULD contain a field with the key `rv`, and a value that corresponds to a 56-bit sampling threshold.
+This value will be compared to the 56-bit random value associated with the trace.
+
+From a technical perspective, how do you propose accomplishing the proposal? In particular, please explain:
+
+* How the change would impact and interact with existing functionality
+* Likely error modes (and how to handle them)
+* Corner cases (and how to handle them)
+
+While you do not need to prescribe a particular implementation - indeed, OTEPs should be about **behaviour**, not implementation! - it may be useful to provide at least one suggestion as to how the proposal *could* be implemented. This helps reassure reviewers that implementation is at least possible, and often helps them inspire them to think more deeply about trade-offs, alternatives, etc.
+
+## Trade-offs and mitigations
+
+This proposal is the result of long negotiations on the Sampling SIG over what is required and various alternative forms of expressing it. [This issue](https://github.com/open-telemetry/opentelemetry-specification/issues/3602) exhaustively covers the various formats that were discussed and their pros and cons. This proposal is the result of that decision.
+
+## Prior art and alternatives
+
+The existing specification for `r-value` and `p-value` attempted to solve this problem, but were limited to powers of 2, which is inadequate.
+
+## Open questions
+
+This specification leaves room for different implementation options. For example, comparing hex strings or converting them to numeric format are both viable alternatives for handling the threshold.
+
+We also know that some implementations prefer to use a sampling probability (in the range from 0-1.0) or a sampling rate (1/probability); this design permits conversion to and from these formats without loss up to at least 6 decimal digits of precision.
+
+## Future possibilities
+
+This permits sampling systems to propagate consistent sampling information downstream where it can be compensated for. For example, this will enable the tail-sampling processor in the OTel Collector to propagate its sampling decisions to backends in a standard way.
```

Comment:
Can you expand a bit on that? Most of the policies in the tail-sampling processor are based on the characteristics of the trace itself. We do have one probabilistic policy, which is probably what you had in mind, but perhaps I'm misunderstanding parts of this proposal. For completeness, you might want to mention the probabilistic sampling processor as well.

Reply:
I intend to revive https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/24811 ASAP, which is I think an example of what Kent is predicting. Of course, our eyes are on more-configurable samplers, but that's future work.

Reply:
I added one line, but if you want more, we can go farther.

The general idea is that we want to be able to pass on the effective sample rate from dynamic sampling decisions, which today's tail sampling processor can't do.